libs/syndesi/src/lib/call.ts

```diff
@@ -0,0 +1,170 @@
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { Resource } from './resource.interfaces';
+import { expand, UriParams } from './uri';
+
+const toNextCall = <S, T>(call: Call<S>) => {
+
+  return function (res: Resource<T> | HttpResponse<Resource<T>>) {
+    const json = res instanceof HttpResponse ? res.body : res;
+    const response = res instanceof HttpResponse  ? res : undefined;
+
+    return new Call<T>(call['_http'], json, response);
+  };
+};
+
+/**
+ * A `Call` is a single HTTP interaction
+ *
+ * ### How To Use
+ *
+ * Inject `ApiClient` and obtain a call instance from a `Resource` obtained prior to this call:
+ *
+ * ```ts
+ * interface Entity {
+ *   whatsUp: string;
+ * }
+ *
+ * @Injectable()
+ * export class EntityService {
+ *   constructor(private api: ApiClient)
+ *
+ *   fetchNextPage(res: Respurce<Entity>) {
+ *     return this.api.call(res).get('next').send<>();
```

Comment:
This is a weak link in the `Call` api. From `<R>send()`, you always get a `Call<R>` with a `Resource<R>`. What about `Call<R, D>` to get a `R extends Resource<D>`?

I think we need:
 - `call()` w/o generics to get an any-typed `Resource<any>`
 - `<R>call()` w/ generic type to get an `R extends Resource<?>`

Thus, the `Call<R extends Resource<?>>` call class would work with Resource....

Reply:
```ts
export class Call2<R> {

  constructor(
    public resource: R & ResourceMetadata
  ) {}

  // PROBLEM: only one of the method declaration overrides seems to work...
  //public next(): Call2<ResourceMetadata>;
  public next<NextResource>(): Call2<NextResource> {
    return new Call2(this.resource);
  }

}

interface FooRes extends Resource {
  bar: string;  
}

const resource1: Resource = {};
const call2 = new Call2({}); // <-- good, compile error

call2.resource._links; // <-- good, type ResourceMetadata found

const next1 = call2.next();
const next2 = call2.next<FooRes>();

next1.resource._links.self; // <-- bad, type ResourceMetadata not recognized
next2.resource._links.self; // <-- good
next2.resource.bar; // <-- good
```

Reply:
In typescript playground, it's supposed to work in TypeScript 3.3